README.MD

```diff
@@ -0,0 +1,326 @@
+    utilityBrandedTokenAddress = utilityBrandedTokenHelper.address;
+});
+
+
+```
+
+##### Deploy GatewayComposer contract
+Gateway composer is a contract optimizes the transactions required to perform the stake and mint process. This contract is deployed on the value chain.
+
+```js
+const gatewayComposerConfig = {
+    deployer: deployerAddress,
+    owner: staker,
+    valueToken: eip20Token,
+    brandedToken: brandedTokenAddress
+};
+deployParams = {
+  gas: '7500000',
+  gasPrice: '0x3B9ACA00'
+};
+const gatewayComposerHelper = Setup.GatewayComposerHelper;
+let gatewayComposerAddress;
+gatewayComposerHelper.setup(gatewayComposerConfig, deployParams).then(function(){
+    gatewayComposerAddress = gatewayComposerHelper.address;
+});
+
+```
+
+##### Deploy Gateway Mock contract
+Gateway contract facilitates the transposition of the BrandedToken from the value chain to the sidechain. The contract is deployed on the value chain.
+
+```js
+let gatewayAddress;
+mockContractDeployerObject.deployMockGatewayPass().then(function(){
+      gatewayAddress = mockContractDeployerObject.addresses.MockGatewayPass;
+});
+
+```
+
+#### BrandedToken Staking through GatewayComposer
+
+##### Approve GatewayComposer for ValueToken & Calls requestStake
+
+```js
+const StakeHelper = BrandedToken.Helpers.StakeHelper;
+const stakeHelperInstance = new StakeHelper(web3Provider, brandedTokenAddress, gatewayComposerAddress);
+let txOptions = {
+  from: owner
+};
+const stakeAmountInWei = '100';
+const stakerGatewayNonce = 1;
+let mintBTAmountInWei;
+stakeHelperInstance.convertToBTToken(stakeAmountInWei, brandedTokenAddress, web3Provider, txOptions).then(function(result){
+    mintBTAmountInWei = result;
+});
+const stakerInstance = new Staker(web3Provider, eip20Token, brandedTokenAddress, gatewayComposerAddress);
+const stakeGasPrice = '0x3B9ACA00';
+const stakeGasLimit = '7500000';
+const beneficiary = '0xaabb1122....................';
+stakerInstance.requestStake(
+    mockTokenAbi,
+    owner,
+    stakeAmountInWei,
+    mintBTAmountInWei,
+    gatewayAddress,
+    stakeGasPrice,
+    stakeGasLimit,
+    beneficiary,
+    stakerGatewayNonce,
+    txOptions
+).then(function(){
+  console.log("Request stake completed!");
+})
+
+```           
+
+##### Facilitator collects worker signature
+The facilitator is a service that facilitates progress of the stake and mint.
+```js
+// StakeRequest nonce stored in BrandedToken contract.
+let stakeRequestNonce;
+// 1. Create TypedData
+const stakeRequestTypedData = stakeHelperInstance.getStakeRequestTypedData(stakeAmountInWei, stakeRequestNonce);
+
+// 2. Generate EIP712 signature object.
+const workerAccountInstance = web3Provider.eth.accounts.wallet[worker];
+workerAccountInstance.signEIP712TypedData(stakeRequestTypedData).then(function(signature){
+  console.log("signature:", signature);
+})
+
+```
+
+##### Facilitator calls acceptStakeRequest
+
+```js
+const Mosaic = require('@openstfoundation/mosaic');
```

Comment:
I believe this should be updated (global comment if there are other instances).

Reply:
@abhayks1

Reply:
Updated with below:
```
const Mosaic = require('@openstfoundation/**mosaic.js**');
```
Let me know if I misunderstood.